django_airavata/apps/api/views.py

```diff
@@ -974,10 +974,12 @@ def put(self, request, format=None):
         data_product_uri = request.query_params['product-uri']
         data_product = request.airavata_client.getDataProduct(
             request.authz_token, data_product_uri)
+        serializer = self.serializer_class(
+            data_product, context={'request': request})
 
         return UserStoragePathView().put(
```

Comment:
Instead of calling `UserStoragePathView().put()`, what if we had either a different version of `user_storage.update_file_content` (maybe called `update_data_product_content`?) or one that took an optional `data_product` argument that would figure out the path internally. `user_storage` already has a `_get_replica_filepath` function - https://github.com/apache/airavata-django-portal-sdk/blob/master/airavata_django_portal_sdk/user_storage.py#L624.

Reply:
This worked @machristie 
Thanks